x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/task/OpenJobPersistentTasksExecutor.java

```diff
@@ -182,7 +184,8 @@ public static String nodeFilter(DiscoveryNode node, Job job) {
                 + "] or higher";
         }
 
-        if (Job.getCompatibleJobTypes(node.getVersion()).contains(job.getJobType()) == false) {
+        if (Job.getCompatibleJobTypes(MlConfigVersion.fromString(node.getAttributes().get(MachineLearning.ML_CONFIG_VERSION_NODE_ATTR)))
```

Comment:
As above, `MlConfigVersion.fromString(node.getAttributes().get(MachineLearning.ML_CONFIG_VERSION_NODE_ATTR))` should be replaced with a call to a new helper method.

And the same for anywhere else where this pattern is needed.